src/main/java/com/ss/ita/rozetka/ui/Modals/BasketModal.java

```diff
@@ -1,37 +1,78 @@
 package com.ss.ita.rozetka.ui.Modals;
 
-import com.codeborne.selenide.ElementsCollection;
+import com.codeborne.selenide.CollectionCondition;
+import com.codeborne.selenide.SelenideElement;
 import com.ss.ita.rozetka.ui.pages.OrderingPage;
-import com.ss.ita.rozetka.ui.pages.ProductPage;
 
-import static com.codeborne.selenide.Selenide.$$x;
-import static com.codeborne.selenide.Selenide.$x;
+import java.util.List;
 
-public class BasketModal {
+import static com.codeborne.selenide.Condition.exist;
+import static com.codeborne.selenide.Condition.text;
+import static com.codeborne.selenide.Selenide.*;
 
-    public ProductPage continueBuying() {
-        $x("//a[contains(@class,'button button_size_medium ')]").click();
-        return new ProductPage();
-    }
+public class BasketModal {
+    // This pattern should only be used in String.format()
+    // with product name as a second argument
+    private static final String PRODUCT_XPATH_PATTERN_FOR_NAME = "//single-modal-window//li[contains(., '%s')]";
 
-    public OrderingPage orderProducts() {
+    public OrderingPage openOrderingPage() {
         $x("//a[contains(@class, 'cart-receipt__submit')]").click();
         return new OrderingPage();
     }
 
-    public String getTotalProductsPriceWithCurrency() {
-        return $x("//div[contains(@class,'sum-price')]").getAttribute("textContent");
+    public int getProductsTotalPrice() {
+        String sum = $x("//div[contains(@class,'sum-price')]//span[1]")
+                .getAttribute("textContent");
+        return Integer.parseInt(sum);
+    }
+
+    public boolean isEmpty() {
+        return $(".cart-dummy").is(exist);
     }
 
-    public int getTotalProductsPrice() {
-        return Integer.parseInt(getTotalProductsPriceWithCurrency().replaceAll("\\D", ""));
+    public List<String> getProductNames() {
+        return $$("li.cart-list__item a.cart-product__title")
+                .shouldHave(CollectionCondition.sizeGreaterThan(0))
+                .texts();
     }
 
-    public boolean isBasketEmpty() {
-        return getProductsList().isEmpty();
+    public BasketModal setProductCount(String productName, int count) {
+        int totalPrice = getProductsTotalPrice();
+
+        String countFieldXpath = String.format(PRODUCT_XPATH_PATTERN_FOR_NAME, productName) +
+                "//input[contains(@class, 'cart-counter__input')]";
```

Comment:
String.format(PRODUCT_XPATH_PATTERN_FOR_NAME + "//input[contains(@class, 'cart-counter__input')]", productName)
same for others